File: DemoServer/AdditionalLanguages/python/queries/pagingQueryResults/pagingQueryResults.py
from typing import Optional
from demo_example import Example, RunParamsBase
from models import Company

#region Usings
from ravendb import QueryStatistics
#endregion

class RunParams(RunParamsBase):
    def __init__(self, results_to_skip: int, results_to_take: int):
        self.results_to_skip = results_to_skip
        self.results_to_take = results_to_take

class PagingQueryResults(Example):
    def run(self, run_params: RunParams):
        resultsToSkip = run_params.results_to_skip
        resultsToTake = run_params.results_to_take

        #region Demo
        with self.document_store_holder.store().open_session() as session:
            stats: Optional[QueryStatistics] = None

            def __stats_callback(statistics: QueryStatistics):
                nonlocal stats
                stats = statistics

            #region Step_1
            paged_query = session.query(object_type=Company)
            #endregion
            
            #region Step_2
            paged_query = paged_query.statistics(__stats_callback)
            #endregion
            #region Step_3
            paged_query.skip(resultsToSkip)
            #endregion
            #region Step_4
            paged_query.take(resultsToTake)
            #endregion
            #region Step_5
            paged_results = list(paged_query)
            #endregion

            #region Step_6
            total_results = stats.total_results
            #endregion
        #endregion

        return paged_results

File: DemoServer/AdditionalLanguages/php/queries/pagingQueryResults/PagingQueryResults.php
<?php

namespace RavenDB\Demo\queries\pagingQueryResults;

//region Usings
use RavenDB\Demo\common\models\Company;
use RavenDB\Documents\Session\QueryStatistics;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;

class PagingQueryResults
{
    public function __invoke(RunParams $runParams)
    {
        $resultsToSkip = $runParams->getResultsToSkip();
        $resultsToTake = $runParams->getResultsToTake();

        //region Demo
        $pagedResults = [];

        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            $statsRef = new QueryStatistics();

            //region Step_1
            $pagedResults = $session->query(Company::class)
            //endregion
                //region Step_2
                ->statistics($statsRef)
                //endregion
                //region Step_3
                ->skip($resultsToSkip)
                //endregion
                //region Step_4
                ->take($resultsToTake)
                //endregion
                //region Step_5
                ->toList();
                //endregion

            //region Step_6
            $totalResults = $statsRef->getTotalResults();
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $pagedResults;
    }
}
